typescript/mainLogic/services/service_context.ts

```diff
@@ -0,0 +1,157 @@
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) { //TODO - Return type for function, typescript doesn't need
+	let NewServiceContext = {
+        client:                                   client,
+        serviceId:                                serviceId,
+        ipAddress:                                ipAddress,
+        testVolumeMountpointOnTestsuiteContainer: testVolumeMountpointOnTestsuiteContainer,
+        testVolumeMountpointOnServiceContainer: testVolumeMountpointOnServiceContainer,
+    }
+    return NewServiceContext; //TODO return pointer in golang, but typescript's reference variables implciltly point into the heap
+}
+
+class ServiceContext { //TODO Should only be adding methods into a class, you can't link to a struct in typescript like in golang
+    
+    client: core_api_bindings_js_grpc.ApiContainerServiceClient;
+    serviceId: ServiceID;
+	ipAddress: string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: core_api_bindings_js_grpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GetServiceID() {
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GetIPAddress() {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public ExecCommand(command: string[]) {
+        let serviceId = this.serviceId;
+        let args = new core_api_bindings_js.ExecCommandArgs();
+        args.setServiceId(serviceId);
+        args.setCommandArgsList(command);
+
+        let resp, err = this.client.execCommand(args); //TODO - no context needed in typescript, so what should I do here?
+        if (err != null) {
+            return [0, null, console.error(
+                err,
+			    "An error occurred executing command '%v' on service '%v'",
+			    command,
+			    serviceId
+            )]; //TODO cannot return multipe values in typescript function, need to wrap in array or object
+        }
+        return resp.ExitCode, resp.LogOutput, null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GenerateFiles(filesToGenerateSet: Map<string, boolean>) {
```

Comment:
1) Same note here as everywhere our functions were capitalized in Go because that's how Go wants it, but Typescript functions should always be lowerCamelCase
2) We do `map[string]bool` in Go because Go doesn't have a formalized Set type (which is stupid), but Typescript actually does so let's use that here to reduce confusion